src/GurobiSolverInterface.jl

```diff
@@ -14,35 +14,48 @@ type GurobiMathProgModel <: AbstractLinearQuadraticModel
     cutcb
     heuristiccb
     infocb
+    options
 end
-function GurobiMathProgModel(;options...)
-   env = Env()
-   setparam!(env, "InfUnbdInfo", 1)
+function GurobiMathProgModel(env=nothing;options...)
+   finalize_env = (env == nothing)
+   if env == nothing
+       env = Env()
+   end
+   m = GurobiMathProgModel(Model(env,""; finalize_env=finalize_env), :Con, false, Float64[], Float64[], nothing, nothing, nothing, nothing, options)
+   # Set the parameters on the model's copy of env rather than modifying the
+   # global env (ref: http://www.gurobi.com/support/faqs#P)
+   setparam!(m.inner, "InfUnbdInfo", 1)
    for (name,value) in options
-       setparam!(env, string(name), value)
+       setparam!(m.inner, string(name), value)
    end
-   m = GurobiMathProgModel(Model(env,""; finalize_env=true), :Con, false, Float64[], Float64[], nothing, nothing, nothing, nothing)
    return m
 end
 
 
 immutable GurobiSolver <: AbstractMathProgSolver
+    env
     options
 end
-GurobiSolver(;kwargs...) = GurobiSolver(kwargs)
-LinearQuadraticModel(s::GurobiSolver) = GurobiMathProgModel(;s.options...)
+GurobiSolver(env=nothing; kwargs...) = GurobiSolver(env, kwargs)
+LinearQuadraticModel(s::GurobiSolver) = GurobiMathProgModel(s.env; s.options...)
 ConicModel(s::GurobiSolver) = LPQPtoConicBridge(LinearQuadraticModel(s))
 
 supportedcones(::GurobiSolver) = [:Free,:Zero,:NonNeg,:NonPos,:SOC,:SOCRotated]
 
 loadproblem!(m::GurobiMathProgModel, filename::AbstractString) = read_model(m.inner, filename)
 
 function loadproblem!(m::GurobiMathProgModel, A, collb, colub, obj, rowlb, rowub, sense)
-  # throw away old model
+  # throw away old model but keep env and finalize_env
   env = m.inner.env
+  finalize_env = m.inner.finalize_env
   m.inner.finalize_env = false
   free_model(m.inner)
-  m.inner = Model(env, "", finalize_env=true)
+  m.inner = Model(env, "", finalize_env=finalize_env)
+  # Re-set options on new model's env
+  setparam!(m.inner, "InfUnbdInfo", 1)
```

Comment:
Another option would be just to set it on the env itself rather than the model's copy?